integreat_compass/cms/views/offers/offer_report_view.py

```diff
@@ -0,0 +1,87 @@
+from django.contrib import messages
+from django.shortcuts import redirect, render
+from django.utils.translation import gettext as _
+from django.views.generic import TemplateView
+
+from ...forms import ReportForm
+from ...models import Offer, OfferVersion
+
+
+class OfferReportView(TemplateView):
+    """
+    View to report an offer
+    """
+
+    template_name = "offers/offer_report.html"
+
+    @staticmethod
+    def _get_form():
+        """
+        This function gets the form Report
+
+        :return: the report form
+        :rtype: dict
+        """
+        form = ReportForm()
+        return form
+
+    def get(self, request, *args, **kwargs):
+        r"""
+        This functions displays all offers
+
+        :param request: The current request
+        :type request: ~django.http.HttpRequest
+
+        :param \*args: The supplied arguments
+        :type \*args: list
+
+        :param \**kwargs: The supplied keyword arguments
+        :type \**kwargs: dict
+
+        :param kwargs: The supplied keyword arguments
+        :type kwargs: dict
+
+        :return: The context of all
+        :rtype: dict
+        """
+        offer = Offer.objects.get(id=kwargs.get("pk"))
+        context = {"selected_offer": offer, "form": self._get_form()}
+        return render(request, self.template_name, context)
+
+    def post(self, request, *args, **kwargs):
+        r"""
+        This functions handles the input from the report function
+
+        :param request: The current request
+        :type request: ~django.http.HttpRequest
+
+        :param \*args: The supplied arguments
+        :type \*args: list
+
+        :param \**kwargs: The supplied keyword arguments
+        :type \**kwargs: dict
+
+        :return: Redirect to the index view
+        :rtype: ~django.http.HttpResponseRedirect
+        """
+        offer = Offer.objects.get(id=kwargs.get("pk"))
+        form = ReportForm(request.POST)
+
+        if not form.is_valid():
+            for form in [*form.values()]:
+                form.add_error_messages(request)
+
+        if form.is_valid():
+            form.instance.offer_version = OfferVersion.objects.get(
+                id=offer.public_version.id
+            )
+            form.save()
+
+        messages.success(
+            request,
+            _(
+                'Your report for offer "{}" has successfully been submitted and will be reviewed shortly.'
+            ).format(offer.public_version.title),
+        )
+
+        return redirect("cms:public:index")
```

Comment:
```suggestion
    def post(self, request, *args, **kwargs):
        r"""
        This functions handles the input from the report function

        :param request: The current request
        :type request: ~django.http.HttpRequest

        :param \*args: The supplied arguments
        :type \*args: list

        :param \**kwargs: The supplied keyword arguments
        :type \**kwargs: dict

        :return: Redirect to the index view
        :rtype: ~django.http.HttpResponseRedirect
        """
        offer = Offer.objects.get(id=kwargs.get("pk"))
        form = ReportForm()

        if not form.is_valid():
            form.add_error_messages(request)
            return render(
                request,
                self.template_name,
                {"form": form, **self.get_context_data(**kwargs)},
            )

        form.instance.offer_version = OfferVersion.objects.get(
            id=offer.public_version.id
        )
        form.save()

        messages.success(
            request,
            _(
                'Your report for offer "{}" has successfully been submitted and will be reviewed shortly.'
            ).format(offer.public_version.title),
        )

        return redirect("cms:public:index")
```

Again, I think the `for form in [*form.values()]:` is a remnant of `OfferFormView`, correct? Since we only have one form here, there's no need to loop. 

I'd also suggest rendering the same view again with the error messages instead of redirecting to index (and showing the success message!) even when an error occurred.

Then you also don't have to check the form validity again.

Reply:
Same here with the handling of non-existant offer PKs.

Reply:
This case I handled through an try-except. I hope this is sufficient

Reply:
With the get_object_or_404 the try-except might be redundant, but I'm not entirely sure, so for now I kept it

Reply:
`get_object_or_404`is a great idea. I don't think the try-except is necessary now, so I'd say it can be removed + we no longer need an explicit else case